### PR TITLE
Fixed the stripping not working the terrain alphatest feature required for terrain holes.

### PR DESCRIPTION
**Pull request by anisunity**
https://fogbugz.unity3d.com/f/cases/1205902/
While trying to implement the fix of the bug, i found out that the stripping didn't work for the terrain alpahtest keyword. This PR fixes it.
I also added a section in terrain lit shader mentioning the message that was requested in the fogbug. Unfortunately it is not possible to add it in the inspector.

# Testing status
Exported with and without the option checked, the stripping works correctly.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader [hunk 1]
```diff
@@ -80,7 +80,7 @@ Shader "HDRP/TerrainLit"
     #pragma multi_compile_instancing
     #pragma instancing_options assumeuniformscaling nomatrices nolightprobe nolightmap
 
-	#pragma multi_compile_local _ _ALPHATEST_ON
+	#pragma multi_compile _ _ALPHATEST_ON
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader:83]: Sorry misclicked on a PR and finished here, but IIRC this change was part of a tentative fix for https://github.com/Unity-Technologies/Graphics/pull/1073 can you verify it is still fine? This register a global keyword which might cause problems.

**Review by JordanL8 — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader (hunk 1 above)
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader:83]: Arf!
summoning @aleks01010101 and @joravainen  for this case. Maybe it hide a bigger issue.
For context, we have issue with multi_compile_local  and stripper, as far as I remember it is know issue, but here we have two opposite "fix" for two different bug, see the link of Francesco: #1073 for detail. THanks

Comment (aleks01010101) [reply com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader:83]: Oh... This bug was assigned to the device grabbag, so we haven't seen it :)

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader:83]: @aleks01010101 do you have more information there?

Comment (aleks01010101) [reply com.unity.render-pipelines.high-definition/Runtime/Material/TerrainLit/TerrainLit.shader:83]: https://fogbugz.unity3d.com/f/cases/1259744/

**Comment by anisunity** [comment]
I just cherry picked the commit on top of master and now the PR targets master. Here is a summary of the situation:
Previously a PR was done by @FrancescoC-unity  #1073 that changed the code to use a local keyword instead of a global one as a workaround for an error message. The issue with that is that we do not handle the alpha_on keyword through the material api for terrains as the terrain material code is not in the regular SRP material one (BaseUnlit, etc). The message error disappeared,  but the stripping was not working properly and that was not catched during the review process.

Recently a c++ PR was merged into 2020.2 and 2021.1 that fixes that message. So that workaround is not required anymore to avoid the error message and we can restore the keyword into the global space.
It would be great to have keyword in the local dmain as the terrain seems to be the only shader that requires it as a keyword instead of a shader_feature, but that would require a c++ PR and would mean trouble if we wanted to rename the keyword. So right now we are going to keep it in the global namesace.

**Review by sebastienlagarde — APPROVED** [review]

**Pull request merged**